Code/GraphMol/Descriptors/GETAWAY.cpp

```diff
@@ -0,0 +1,804 @@
+                  tmpi = getH((double)Wi(j), (double)Wi(k), (double)R(j,k)); // Use same function but on all R not "H>0" like in the previous loop & i>0!
+                  tmps = getH((double)Ws(j), (double)Ws(k), (double)R(j,k)); // Use same function but on all R not "H>0" like in the previous loop & i>0!
+                  R0u+=tmpu;
+                  R0m+=tmpm;
+                  R0v+=tmpv;
+                  R0e+=tmpe;
+                  R0p+=tmpp;
+                  R0i+=tmpi;
+                  R0s+=tmps;
+                  if (tmpu>Rkmaxu) {
+                    Rkmaxu=tmpu;
+                  }
+                  if (tmpm>Rkmaxm) {
+                    Rkmaxm=tmpm;
+                  }
+                  if (tmpv>Rkmaxv) {
+                    Rkmaxv=tmpv;
+                  }
+                  if (tmpe>Rkmaxe) {
+                    Rkmaxe=tmpe;
+                  }
+                  if (tmpp>Rkmaxp) {
+                    Rkmaxp=tmpp;
+                  }
+                  if (tmpi>Rkmaxi) {
+                    Rkmaxi=tmpi;
+                  }
+                  if (tmps>Rkmaxs) {
+                    Rkmaxs=tmps;
+                  }
+              }
+            }
+          }
+          Rk[0][i-1]=R0u;
+          Rk[1][i-1]=R0m;
+          Rk[2][i-1]=R0v;
+          Rk[3][i-1]=R0e;
+          Rk[4][i-1]=R0p;
+          Rk[5][i-1]=R0i;
+          Rk[6][i-1]=R0s;
+
+          Rp[0][i-1]=Rkmaxu;
+          Rp[1][i-1]=Rkmaxm;
+          Rp[2][i-1]=Rkmaxv;
+          Rp[3][i-1]=Rkmaxe;
+          Rp[4][i-1]=Rkmaxp;
+          Rp[5][i-1]=Rkmaxi;
+          Rp[6][i-1]=Rkmaxs;
+
+        }
+  }
+
+  //HATSk[0]+2*(HATSk[1]+HATSk[2]+HATSk[3]+HATSk[4]+HATSk[5]+HATSk[6]+HATSk[7]+HATSk[8]);
+  double HATSTu = getHtotal(HATSk[0]);
+  double HATSTm = getHtotal(HATSk[1]);
+  double HATSTv = getHtotal(HATSk[2]);
+  double HATSTe = getHtotal(HATSk[3]);
+  double HATSTp = getHtotal(HATSk[4]);
+  double HATSTi = getHtotal(HATSk[5]);
+  double HATSTs = getHtotal(HATSk[6]);
+
+
+  // Hk[0]+2*(Hk[1]+Hk[2]+Hk[3]+Hk[4]+Hk[5]+Hk[6]+Hk[7]+Hk[8]);
+  double HTu = getHtotal(Hk[0]);
+  double HTm = getHtotal(Hk[1]);
+  double HTv = getHtotal(Hk[2]);
+  double HTe = getHtotal(Hk[3]);
+  double HTp = getHtotal(Hk[4]);
+  double HTi = getHtotal(Hk[5]);
+  double HTs = getHtotal(Hk[6]);
+
+  //
+  //2*(Rk[1]+Rk[2]+Rk[3]+Rk[4]+Rk[5]+Rk[6]+Rk[7]+Rk[8]);
+  double RTu = getRtotal(Rk[0]);
+  double RTm = getRtotal(Rk[1]);
+  double RTv = getRtotal(Rk[2]);
+  double RTe = getRtotal(Rk[3]);
+  double RTp = getRtotal(Rk[4]);
+  double RTi = getRtotal(Rk[5]);
+  double RTs = getRtotal(Rk[6]);
+
+
+  double RTMu=getMax(Rp[0]);
+  double RTMm=getMax(Rp[1]);
+  double RTMv=getMax(Rp[2]);
+  double RTMe=getMax(Rp[3]);
+  double RTMp=getMax(Rp[4]);
+  double RTMi=getMax(Rp[5]);
+  double RTMs=getMax(Rp[6]);
+
+  // create the output vector...
+  for (int i=0;i<9;i++){
+    res.push_back(Hk[0][i]);
+   }
+  res.push_back(HTu);
```

Comment:
I can see why you structure the code like this. However it separates the determination of Hk and HTu from the assignment to the result vector and so on. Some components of the result vector are already assigned above, keeping them closer to the point where they are calculated. Would it be better for longer term maintenance to move calculation and assignment to result closer together?